packages/xchain-bitcoincash/src/client.ts

```diff
@@ -286,7 +286,7 @@ class Client extends UTXOClient {
 
     const keyPair = this.getBCHKeys(this.phrase, derivationPath)
 
-    utxos.forEach((utxo, index) => {
```

Comment:
here it tried to sign every utxo found at the address, although some are possibly not chosen for the inputs array